### PR TITLE
Fix #2294 - drop HPO term collection later into update

### LINKED ISSUE #2294: Update hpo command crashes
It's not the first time that this happens when I'm testing something locally. Today it happens again so I think we should revise the code and force to first download the files, then parsing these files and save stuff to the database and finally removing the files from disk.

Current error:

image

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. test updating HPO collection with working and missing/broken files (or dl, but..)

**Expected outcome**:
The functionality should be working. Collection should not drop when missing files even if the user has acknowledged a drop may occur.
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
```
(scout39n) ➜  scout_dl_230419 scout update hpo  --hpoterms hpo.obo --hpo-to-genes phenotype_to_genes.txt
2023-08-18 11:42:38 MacBook-Pro-3.local scout.commands.base[75488] INFO Running scout version 4.67
2023-08-18 11:42:38 MacBook-Pro-3.local scout.commands.base[75488] DEBUG Debug logging enabled.
2023-08-18 11:42:38 MacBook-Pro-3.local scout.adapter.client[75488] INFO Connected to MongoDB 6.0.1
2023-08-18 11:42:38 MacBook-Pro-3.local scout.adapter.mongo.base[75488] DEBUG Database name: scout
2023-08-18 11:42:38 MacBook-Pro-3.local scout.server.app[75488] INFO Enable Phenopacket API
Are you sure you want to drop the hpo terms? [y/N]: y
2023-08-18 11:42:45 MacBook-Pro-3.local scout.commands.update.hpo[75488] INFO Running Scout update HPO
2023-08-18 11:42:45 MacBook-Pro-3.local scout.load.hpo[75488] INFO Parsing hpo terms
2023-08-18 11:42:46 MacBook-Pro-3.local scout.adapter.mongo.hgnc[75488] INFO Fetching all genes by alias
2023-08-18 11:42:48 MacBook-Pro-3.local scout.load.hpo[75488] INFO Dropping old HPO term collection
Loading hpo terms  [------------------------------------]    0%2023-08-18 11:42:49 MacBook-Pro-3.local scout.build.hpo[75488] DEBUG Building hpo term HP:0000001
```

File: scout/load/hpo.py [hunk 1]
```diff
@@ -118,6 +118,13 @@ def load_hpo_terms(
 
         hpo_term["genes"].add(hgnc_id)
 
+    if not hpo_terms:
```
Comment (northwestwitch) [inline scout/load/hpo.py:121]: 👍🏻

File: scout/load/hpo.py [hunk 2]
```diff
@@ -118,6 +118,13 @@ def load_hpo_terms(
...
+        LOG.error("No HPO terms fogund. Aborting update without dropping HPO term collection.")
```
Comment (northwestwitch) [inline scout/load/hpo.py:122]:
```suggestion
        LOG.error("No HPO terms found. Aborting update without dropping HPO term collection.")
```

**Review by northwestwitch — APPROVED** [review]
Nice and effective! 🥇

File: scout/load/hpo.py (hunk 2 above)
Comment (dnil) [reply scout/load/hpo.py]: Ahem, just one more of those editor focus things I'm sure. 😁

Comment (northwestwitch) [reply scout/load/hpo.py]: No worries, that's what the reviews are for! 😄

**Pull request merged**